CONTRIBUTING.md

```diff
@@ -36,3 +32,27 @@ Additionally, for the sake of simplicity, we are standardizing on what should be
 
 ![screen shot 2016-08-16 at 8 03 22 pm](https://cloud.githubusercontent.com/assets/777633/17719999/920fb534-63ec-11e6-8903-3725f2cd05b0.png)
 ![screen shot 2016-08-16 at 8 03 36 pm](https://cloud.githubusercontent.com/assets/777633/17719998/920cff6a-63ec-11e6-8f76-0ac7a5fa0c9d.png)
+
+## Best Practices for Contributing
+[Best Practices for Contributing]: #best-practices-for-contributing
+* Before you start coding, open an issue so that the community can discuss your change to ensure it is in line with the goals of the project and not being worked on by someone else. This allows for discussion and fine tuning of your feature and results in a more succent and focused additions.
+
+* Pull Requests represent final code. Please ensure they are:
+     * Well tested by the author. It is the author's job to ensure their code works as expected.  
+     * Be free of unnessicary log calls. Logging is great for debugging, but when a PR is made, log calls should only be present when there is an actual error or to warn of an unimplemented feature.
+   
+   If your code is untested or log heavy, prefix your PR with "WIP", so others know it is still being tested and shouldn't be considered for merging yet.
+
+* Small changes are preferable over large ones. The larger a change is the more likely it is to conflict with the project and thus be denied. If your additon is large, be sure to extensively discuss it in an issue request before you submit a PR or even start coding.
```

Comment:
Small spelling error here as well, "additon vs addition". "Issue request" should just be "Issue", missing a comma between "project" and "and", as well as"PR" and "or".